### PR TITLE
7570: Drop JasperDB (#7803)

### PR DESCRIPTION
**Pull request by imalygin**
This is the second attempt for this PR -https://github.com/hashgraph/hedera-services/pull/7803
Now `hedera-node` is ready as well. 

Fix summary:

- All JasperDB sources are removed
- MerkleDb sources are still in swirlds-jasperdb folder to minimize changes
- Java module is renamed from swirlds-jasperdb to swirlds-merkledb
- JasperDB variations of PTT and other unit tests are removed
- Many classes that had "MerkleDb" suffix are now dropped that suffix (e.g. VirtualMapMerkleDbReconnectTest -> VirtualMapReconnectTest)

Fixes: https://github.com/hashgraph/hedera-services/issues/7570

### CODE REVIEW

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/virtualmerkle/map/smartcontracts/data/SmartContractMapKeySerializer.java [hunk 1]
```diff
@@ -16,65 +16,31 @@
...
+        super(CLASS_ID, ClassVersion.ORIGINAL, SmartContractMapKey.getSizeInBytes(), 1);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public int deserializeKeySize(final ByteBuffer buffer) {
-        return SmartContractMapKey.getSizeInBytes();
+    protected SmartContractMapKey newKey() {
```
Comment (artemananiev) [inline platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/virtualmerkle/map/smartcontracts/data/SmartContractMapKeySerializer.java:42]: I wonder why it isn't passed as an extra arg to the super constructor?
```java
super(CLASS_ID, ClassVersion.ORIGINAL, SmartContractMapKey.getSizeInBytes(), 1, SmartContractMapKey::new);
```

**Review by artemananiev — DISMISSED** [review]

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/virtualmerkle/map/smartcontracts/data/SmartContractMapKeySerializer.java (hunk 1 above)
Comment (imalygin) [reply platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/virtualmerkle/map/smartcontracts/data/SmartContractMapKeySerializer.java]: For no good reason. I did as you suggested and removed the method.

**Comment by imalygin** [comment]
This is a follow up PR for `hedera-node` repo: https://github.com/hashgraph/hedera-services/pull/7968

**Review by jeromy-cannon — APPROVED** [review]

**Review by jsync-swirlds — APPROVED** [review]
Pretty straight-forward, albeit long.
I see no major concerns, but thorough testing (including services and JRS/FST) is needed.

**Review by artemananiev — APPROVED** [review]

**Review by poulok — APPROVED** [review]

**Pull request merged**